src/lib.rs

```diff
@@ -13,6 +13,12 @@
 
 #![cfg_attr(feature = "nightly", feature(drain))]
 
+// optional serde support
+#![cfg_attr(feature = "serde_macros", feature(const_fn, custom_derive, plugin))]
```

Comment:
Sorry, I realize what I wrote as ambiguous. What I mean is, I'd expect the line to look like

```rust
#![cfg_attr(feature = "eders", feature(const_fn, custom_derive, plugin))]
```

because that's the Cargo feature the user would enable in order to turn on `serde` support. (And yes, that will only work on nightly.) The other `cfg` and `cfg_attr` attributes should do the same.